tests/Integration/inc/Engine/Cache/AdvancedCache/updateAdvancedCache.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace WP_Rocket\Tests\Integration\inc\Engine\Cache\AdvancedCache;
+
+use WP_Rocket\Engine\Cache\AdvancedCache;
+use WP_Rocket\Tests\Integration\FilesystemTestCase;
+use Brain\Monkey\Filters;
+
+/**
+ * @covers \WP_Rocket\Engine\Cache\AdvancedCache::update_advanced_cache
+ * @uses   ::rocket_get_filesystem_perms
+ *
+ * @group  AdvancedCache
+ */
+class Test_UpdateAdvancedCache extends FilesystemTestCase {
+	protected $path_to_test_data = '/inc/Engine/Cache/AdvancedCache/updateAdvancedCache.php';
+
+	public function testShouldBailOutWhenShortCircuitFilterReturnsFalse() {
```

Comment:
You can add this test to the test_data too, the same way as described in my other comment.

For the filter, you can use a config value from the data provider, and do the following:
```
if ( false === $config['generate'] ) {
    add_filter( 'rocket_generate_advanced_cache_file', [ $this, 'return_false' ] );
}
```

We have custom methods for tests to return common values like this. Also since we're adding the filter in the test, we want to remove it on the `tearDown()` method to make sure it doesn't interfere with other tests

```
public function tearDown() {
    parent::tearDown();
    remove_filter( 'rocket_generate_advanced_cache_file', [ $this, 'return_false' ] );
}
```